specs/configure_monitoring.adoc

```diff
@@ -42,22 +42,159 @@ The use cases addressed by this spec are:
 * Tendrl core stack is installed and one/more nodes are managed by tendrl core.
   Now the user decides to monitor his tendrl managed nodes by installing the
   tendrl/performance_monitoring.
+* User plugs out the monitoring after using for a while.
 
 == Proposed change
 
-* The tendrl/performance_monitoring on start, uploads the default monitoring
-  configuration and flows for generating the collectd configurations.
+* The tendrl/performance_monitoring on start, checks the existence of
+  '/monitoring/defaults' directory in etcd and if its not present, it uploads
+  the default monitoring configuration and flows for generating the collectd
+  configurations.
 * The performance_monitoring application then looks into the etcd's '/nodes'
-  directory to see if there are any nodes that are currently managed by tendrl
+  directory to see if there are any nodes that are currently managed by tendrl.
   If there are any, it triggers the monitoring provisioning flow on every node
   for node monitoring via the core API.
   ** The provisioning flow for the monitoring stack installs collectd-related
      pieces on every node and generates the configuration files from templates
      shipped with the collectd plugins on nodes.
 * For any new node that is managed by tendrl, the performance_monitoring
-  application repeats the above procedure of intiating the monitoring
-  provisioning flow.
+  application invokes a core api to intiate the monitoring provisioning flow
+  on that node.This api then loads a job with the api parameters as job
+  parameters along with the fqdn of the new node. The job is then executed by
+  node-agent residing on that node.
+* If the user decides not to have monitoring anymore, an api will be exposed
+  by the core API which will internally invoke a flow that does the following:
+  **  Stop collectd on every tendrl managed node
+  **  Stop performance-monitoring application
+  This means that even though the monitoring is turned off, the actual
+  monitoring related bits are only terminated from service and not removed from
+  existence
```

Comment:
I think we can de-prioritise the turn-off mechanism for now and concentrate on the installation and use of the monitoring stack. I would rather not have the turn-off mechanism take any time away for now.